lib/request-base.js

```diff
@@ -130,6 +130,50 @@ RequestBase.prototype.timeout = function timeout(options){
 };
 
 /**
+ * Set number of retry attempts on error.
+ *
+ * Failed requests will be retried 'count' times if timeout or err.code >= 500.
+ *
+ * @param {Number} count
+ * @return {Request} for chaining
+ * @api public
+ */
+
+RequestBase.prototype.retry = function retry(count){
+  this._maxRetries = count || 1;
+  this._retries = 0;
```

Comment:
I'm not sure I follow. This is the main api for enabling `retry` (like `timeout`). I tried to follow the same logic as for redirects.